test/e2e/subscription_e2e_test.go

```diff
@@ -612,3 +614,149 @@ func TestSusbcriptionWithStartingCSV(t *testing.T) {
+	var wg sync.WaitGroup
+	go func(t *testing.T) {
+		wg.Add(1)
+		defer wg.Done()
+		_, err := awaitCSV(t, crc, testNamespace, csvB.GetName(), csvAnyChecker)
+		require.NoError(t, err)
+	}(t)
+	// Update the catalog to include multiple updates
+	packages := []registry.PackageManifest{
+		{
+			PackageName: packageName,
+			Channels: []registry.PackageChannel{
+				{Name: stableChannel, CurrentCSVName: csvC.GetName()},
+			},
+			DefaultChannelName: stableChannel,
+		},
+	}
+
+	updateInternalCatalog(t, c, crc, catalogSourceName, testNamespace, []apiextensions.CustomResourceDefinition{crd}, []v1alpha1.ClusterServiceVersion{csvA, csvB, csvC}, packages)
+
+	// wait for checks on intermediate csvs to succeed
+	wg.Wait()
+
+	// Wait for csvC to be installed
+	_, err = awaitCSV(t, crc, testNamespace, csvC.GetName(), csvSucceededChecker)
+	require.NoError(t, err)
+
+	// Should eventually GC the CSVs
+	err = waitForCSVToDelete(t, crc, csvA.Name)
+	require.NoError(t, err)
+	err = waitForCSVToDelete(t, crc, csvB.Name)
+	require.NoError(t, err)
+
+	// TODO: check installplans, subscription status, etc
+}
+
+func updateInternalCatalog(t *testing.T, c operatorclient.ClientInterface, crc versioned.Interface, catalogSourceName, namespace string, crds []apiextensions.CustomResourceDefinition, csvs []v1alpha1.ClusterServiceVersion, packages []registry.PackageManifest) {
+	fetchedInitialCatalog, err := fetchCatalogSource(t, crc, catalogSourceName, namespace, catalogSourceRegistryPodSynced)
+	require.NoError(t, err)
+
+	// Get initial configmap
+	configMap, err := c.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Get(fetchedInitialCatalog.Spec.ConfigMap, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	// Update package to point to new csv
+	manifestsRaw, err := yaml.Marshal(packages)
+	require.NoError(t, err)
+	configMap.Data[registry.ConfigMapPackageName] = string(manifestsRaw)
+
+	// Update raw CRDs
+	var crdsRaw []byte
+	crdStrings := []string{}
+	for _, crd := range crds {
+		crdStrings = append(crdStrings, serializeCRD(t, crd))
+	}
+	crdsRaw, err = yaml.Marshal(crdStrings)
+	require.NoError(t, err)
+	configMap.Data[registry.ConfigMapCRDName] = strings.Replace(string(crdsRaw), "- |\n  ", "- ", -1)
+
+	// Update raw CSVs
+	csvsRaw, err := yaml.Marshal(csvs)
+	require.NoError(t, err)
+	configMap.Data[registry.ConfigMapCSVName] = string(csvsRaw)
+
+	// Update configmap
+	_, err = c.KubernetesInterface().CoreV1().ConfigMaps(testNamespace).Update(configMap)
+	require.NoError(t, err)
+
+	// wait for catalog to update
+	_, err = fetchCatalogSource(t, crc, catalogSourceName, testNamespace, func(catalog *v1alpha1.CatalogSource) bool {
+		if catalog.Status.LastSync != fetchedInitialCatalog.Status.LastSync && catalog.Status.ConfigMapResource.ResourceVersion != fetchedInitialCatalog.Status.ConfigMapResource.ResourceVersion {
+			fmt.Println("catalog updated")
+			return true
+		}
+		fmt.Println("waiting for catalog pod to be available")
```

Comment:
The gRPC pod may still take some time before it spins up after the `CatalogSource` is synced, right?